File: apps/api/app/schemas/exam.py
"""Exam schemas"""

from datetime import date, datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict, Field


class ExamBase(BaseModel):
    name: str = Field(..., max_length=255)
    score: Optional[str] = Field(None, max_length=100)
    exam_date: Optional[date] = None
    description: Optional[str] = None


class ExamCreate(ExamBase):
    pass


class ExamUpdate(BaseModel):
    name: Optional[str] = Field(None, max_length=255)
    score: Optional[str] = Field(None, max_length=100)
    exam_date: Optional[date] = None
    description: Optional[str] = None


class ExamResponse(ExamBase):
    id: str
    user_id: str
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: apps/web/types/exam.ts
export interface Exam {
  id: string;
  user_id: string;
  name: string;
  score: string | null;
  exam_date: string | null;
  description: string | null;
  created_at: string;
  updated_at: string;
}

export interface ExamCreate {
  name: string;
  score?: string | null;
  exam_date?: string | null;
  description?: string | null;
}

export type ExamUpdate = Partial<ExamCreate>;

